modules/imap/hm-imap.php

```diff
@@ -1099,7 +1099,7 @@ public function search($target='ALL', $uids=false, $terms=array(), $esearch=arra
             if ($only_auto_bcc) {
                $fld .= ' HEADER X-Auto-Bcc cypht';
             }
-            if (!mb_strstr($this->server, 'yahoo') && $exclude_auto_bcc) {
+            if ($this->server_supports_custom_headers() && !mb_strstr($this->server, 'yahoo') && $exclude_auto_bcc) {
```

Comment:
Note that this adds one more IMAP query to every search query which will have a performance impact. I suggest you reverse the order of checks here, so it is only called when needed - e.g. $exclude_auto_css && ! mb_strstr && $this->server_supports_custom_headers...

Reply:
Done